lib/python/rose/apps/rose_ana.py

```diff
@@ -48,6 +50,85 @@
 USRCOMPARISON_EXT = ".py"
 
 
+class KGODatabase(object):
+    """
+    KGO Database object, stores comparison information for rose_ana apps.
+
+    """
+    def __init__(self):
+        """Initialise the object"""
+        self.file_name = os.path.join(
+            os.getenv("ROSE_SUITE_DIR"), "log", "rose-ana-comparisons.db")
+        self.conn = None
+        self.create()
+
+    def get_conn(self):
+        """Return a connection to the database if it exists"""
+        if self.conn is None:
+            self.conn = sqlite3.connect(self.file_name, timeout=60.0)
+        return self.conn
+
+    def create(self):
+        """If the databaste doesn't exist, create it"""
+        self.wait_for_lock()
+        if not os.path.exists(self.file_name):
+            conn = self.get_conn()
+            # This table stores the individual comparisons
+            conn.execute(
+                """
+                CREATE TABLE comparisons (
+                app_task TEXT,
+                kgo_file TEXT,
+                suite_file TEXT,
+                status TEXT,
+                comparison TEXT,
+                PRIMARY KEY(app_task))
+                """)
+            # And this one stores the completion status of the task
+            conn.execute(
+                """
+                CREATE TABLE tasks (
+                task_name TEXT,
+                completed INT,
+                PRIMARY KEY(task_name))
+                """)
+            conn.commit()
+        self.unlock()
+
+    def wait_for_lock(self):
```

Comment:
Yes, that's right - the locking is for creation as multiple tasks trying to create the database could be bad.  The updates to the database are protected by sqlite3's own locking mechanism (which has a timeout I've set to 60 seconds)